server/rpc.go

```diff
@@ -264,6 +284,11 @@ func (h *lphttp) ServeSegment(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	err = processPayment(orch, r.Header.Get(PaymentHeader), segData.ManifestID)
+	if err != nil {
+		glog.Errorf("Error processing payment: %v", err)
```

Comment:
We continue to process the segment even if the payment fails?

Reply:
I agree with @j0sh point about sending an error response and exiting if the node is in *on-chain* mode. That being said, could there be a problem if we exit early as is when the node is in *off-chain* mode when the broadcaster tries to unmarshal an empty buffer in `processPayment`. If so, then the orchestrator will always exit early. Perhaps there should also be a check in `processPayment` to return nil if the node is in off-chain mode

Reply:
IIRC, the deserialized `payment` that we receive from protobufs will be empty (probufs won't return an error on missing fields since all fields are optional in protobufs v3). The `orch.ProcessPayment` check does incorporate an off-chain check, so we should be all set here.

Would be good to test in `-offchain` mode just to be sure, though.

Reply:
Ah nice. Might be worthwhile to add an extra test with an empty payment string header too

Reply:
Awesome feedback - changed the code accordingly!